lib/std/private/shlexutils.nim

```diff
@@ -0,0 +1,83 @@
+type State = enum
+  sInStart
```

Comment:
So ... what is this based on? Where can I read the spec and ensure you got this state machine right? "It's now bug compatible with Python as it was translated from Python" is not good enough. Previously we were bug compatible with our simpler implementation.

Reply:
> Previously we were bug compatible with our simpler implementation.

that simpler implementation was not following any spec, and was a very poor approximation to how the shell parses inputs, failing in simple common cases as mentioned in https://github.com/nim-lang/Nim/issues/14343, irrespective of which shell is being used; I'm not talking shell specific edge cases but about behavior in all posix shells regarding how quoting works with single quotes, double quotes, and backslash.

Furthermore, it failed to satisfy `a.quoteShellCommand.parseCmdLine == a`, unlike this PR.

That said, I've now added a commit to make backslash escaping work according to spec, as followed by most if not all shells, see some example specs here:
* https://bash.cyberciti.biz/guide/Quoting
* https://ss64.com/bash/syntax-quoting.html
* https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_03_03.html

It's also consistent with how python3 works, with `list(shlex.shlex(..., posix=True)))`